AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginLazyLoadTests/LL1/AWSDataStoreLazyLoadPostComment4V2StressTests.swift

```diff
@@ -0,0 +1,211 @@
+    ///         - the post should be synced before the comment
+    ///         - it should not skip over an item, ie. a comment saved but post is missing.
+    ///     - The remote store should contain all items synced
+    ///         - comments and post should exist.
+    ///         - the comment should also have the post reference.
+    ///
+    func testMultiSaveWithInterruptions() async throws {
+        await setup(withModels: PostComment4V2Models())
+        let amplify = AmplifyTestExecutor()
+
+        Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] Begin saving data with interruptions")
+        let savesSyncedExpectation = expectation(description: "Outbox is empty after saving (with interruptions)")
+        savesSyncedExpectation.assertForOverFulfill = false
+        try await amplify.multipleSavesWithInterruptions(savesSyncedExpectation)
+        await fulfillment(of: [savesSyncedExpectation], timeout: 120)
+
+        Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] Outbox is empty, begin asserting data")
+        let savedModels = await amplify.savedModels
+        for savedModel in savedModels {
+            let savedComment = savedModel.0
+            let savedPost = savedModel.1
+
+            try await assertQueryComment(savedComment, post: savedPost)
+            try await assertQueryPost(savedPost)
+        }
+        Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] All models match remote store, begin clean up.")
+        try await cleanUp(savedModels)
+    }
+
+    actor AmplifyTestExecutor {
+        var savedModels = [(Comment, Post)]()
+
+        /// The minimum number of iterations, through trial and error, found to reproduce the bug.
+        private let count = 15
+        
+        /// `isOutboxEmpty` is used to return the flow back to the caller via fulfilling the `savesSyncedExpectation`.
+        /// By listening to the OutboxEvent after performing the operations, the last outboxEvent to be `true` while `index`
+        /// is the last index, will be when `savesSyncedExpectation` is fulfilled and returned execution back to the caller.
+        private var isOutboxEmpty = false
+
+        private var index = 0
+        private var subscribeToOutboxEventTask: Task<Void, Never>?
+        private var outboxEventsCount = 0
+
+        /// Perform saving the comment/post in one detached task while another detached task will
+        /// perform the interruption (stop/start). Repeat with a bit of delay to allow DataStore some
+        /// time to kick off its start sequence- this will always be the case since the last operation of
+        /// each detached task is a `save` (implicit `start`) or a `start`.
+        func multipleSavesWithInterruptions(_ savesSyncedExpectation: XCTestExpectation) async throws {
+            subscribeToOutboxEvent(savesSyncedExpectation)
+            while isOutboxEmpty == false {
```

Comment:
why do we need this block? i am guessing `isOutboxEmpty` will always be true when we start the test?